rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/terminal/LSPTerminalREPL.java

```diff
@@ -104,25 +104,21 @@ protected IDEServices buildIDEService(PrintWriter err, IRascalMonitor monitor, T
     @Override
     protected Evaluator buildEvaluator(Reader input, PrintWriter stdout, PrintWriter stderr, IDEServices services) {
         var evaluator = super.buildEvaluator(input, stdout, stderr, services);
-        evaluator.addRascalSearchPath(URIUtil.correctLocation("lib", "rascal-lsp", ""));
-
-        URIResolverRegistry reg = URIResolverRegistry.getInstance();
+        try {
+            ISourceLocation projectDir = PathConfig.inferProjectRoot(URIUtil.createFileLocation(System.getProperty("user.dir")));
```

Comment:
Add a comment that this user.dir business is due to the fact that we start the terminal process with the project folder as user dir.

Complexer code can be replaced by root location("cwd")

Reply:
Except that on `cwd` we cannot go to the parent? I was wondering the same, but I wanted to keep the code as close as possible to the original logic.

Fix that at a later point, if we have all this working.

Reply:
Cwd is logical and will be resolved to a file uri before the parent is found. The code that checks the folder name to be equal to the project name takes care of this in createPathConfig